test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -283,7 +298,11 @@ protected void configure() {
                         bind(NamedWriteableRegistry.class).toInstance(namedWriteableRegistry);
                     }
                 }
-        ).createInjector();
+        );
+        pluginsService.processModules(modulesBuilder);
+        injector = modulesBuilder.createInjector();
+        IndexScopedSettings indexScopedSettings = injector.getInstance(IndexScopedSettings.class);
+        idxSettings = IndexSettingsModule.newIndexSettings(index, indexSettings, indexScopedSettings);
```

Comment:
I need it after the guice injector has been created, so that I can get `IndexScopedSettings` instance. The percolator query builder test that tests the `doToQuery` would otherwise fail, since there an index setting is checked that the `PercolatorPlugin` registers.